### PR TITLE
Show warning to confirm credit card from settings

### PR DESCRIPTION
**Pull request by Betree**
Require https://github.com/opencollective/opencollective-api/pull/8268
Related https://github.com/opencollective/opencollective/issues/6249
Resolve https://github.com/opencollective/opencollective/issues/6250
Resolve https://github.com/opencollective/opencollective/issues/6223

Also moved queries to GQLV2.

https://user-images.githubusercontent.com/1556356/209813789-d22883cc-dfdc-45ec-bc78-7a029642e1cc.mp4

### CODE REVIEW

File: components/edit-collective/EditPaymentMethod.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import { Box, Flex } from '../Grid';
+import Link from '../Link';
+import StyledButton from '../StyledButton';
+
+type EditPaymentMethodProps = {
+  collectiveSlug: string;
+  isSaving: boolean;
+  needsConfirmation: boolean;
+  nbActiveSubscriptions: number;
+  paymentMethod: any; // TODO Replace with PaymentMethod type once parent will be typed
```
Comment (kewitz) [inline components/edit-collective/EditPaymentMethod.tsx:16]: Can't we fetch this from the types being automatically generated?

File: components/edit-collective/sections/PaymentMethods.js [hunk 2]
```diff
@@ -211,14 +286,55 @@ class EditPaymentMethods extends React.Component {
...
+                    id="Order.Confirm.Success"
+                    defaultMessage="Your payment method has now been confirmed and the payment successfully went through."
+                  />
+                ) : (
+                  <FormattedMessage defaultMessage="Your payment method has been successfully added." />
+                )}
+              </span>
+              <StyledButton buttonSize="tiny" onClick={this.dismissSuccess}>
+                <FormattedMessage defaultMessage="Dismiss" />
+              </StyledButton>
```
Comment (kewitz) [inline components/edit-collective/sections/PaymentMethods.js:334]: Should we use the ✖️ button?

**Review by kewitz — APPROVED** [review]

File: components/edit-collective/sections/PaymentMethods.js (hunk 2 above)
Comment (Betree) [reply components/edit-collective/sections/PaymentMethods.js:334]: This is how it appears in the design:

image

File: components/edit-collective/EditPaymentMethod.tsx (hunk 1 above)
Comment (Betree) [reply components/edit-collective/EditPaymentMethod.tsx:16]: Absolutely, I was thinking that we needed for the query in `components/edit-collective/sections/PaymentMethods.js` to be typed but we already have what we need in `components/edit-collective/sections/PaymentMethods.js`. Pushing an update.

**Pull request merged**